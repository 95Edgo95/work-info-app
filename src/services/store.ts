import createSagaMiddleware, {SagaMiddleware} from "redux-saga";
import {applyMiddleware, compose, createStore} from "redux";
import reducer from "services/reducer";
import saga from "services/saga";
import {fromJS} from "immutable";

let store: any;

const initialState: any = fromJS({});

const sagaMiddleware: SagaMiddleware<any> = createSagaMiddleware();

if (process.env && process.env.NODE_ENV !== "production") {
    const composeEnhancers: any = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    store = createStore(reducer, initialState, composeEnhancers(applyMiddleware(sagaMiddleware)));
} else {
    store = createStore(reducer, initialState, compose(applyMiddleware(sagaMiddleware)));
}

sagaMiddleware.run(saga);

export default store;
