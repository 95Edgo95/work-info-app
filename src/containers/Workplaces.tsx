import {attemptDeleteWorkplace, attemptGetWorkplaces} from "store/workplace/WorkplaceActionCreators";
import {mapDispatchToProps, mapStateToProps} from "helpers/ReduxHelper";
import {IWorkbook} from "store/workbook/WorkbookReducer";
import {ISelectorVariables} from "services/selector";
import EntityActions from "components/EntityActions";
import CardsWrapper from "containers/CardsWrapper";
import {RouteComponentProps} from "react-router";
import Filter from "components/Filter";
import {connect} from "react-redux";
import * as React from "react";
import {Map} from "immutable";

interface IWorkplaceReduxProps {
  workplaces: Map<string, IWorkbook>;
  isLoading: boolean;
  pagination: any;
  role: string;
}

interface IWorkplaceDispatchProps {
  attemptDeleteWorkplace: (id: string) => void;
  attemptGetWorkplaces: (params: any) => void;
}

interface IWorkplaceProps extends IWorkplaceReduxProps, IWorkplaceDispatchProps, RouteComponentProps<any> {}

function Workplace(props: IWorkplaceProps) {
  const {
    match: {params: {workbookId}},
    attemptDeleteWorkplace,
    attemptGetWorkplaces,
    workplaces,
    pagination,
    isLoading,
    history,
    role,
  } = props;

  return (
    <React.Fragment>
      <Filter
        attemptGetEntities={attemptGetWorkplaces}
        createLink={`workplaces/${workbookId}`}
        additionalParams={{ workbookId }}
        showAddNew={role === "admin"}
        inputs={[
          {
            placeholder: "Country",
            defaultValue: "",
            name: "country",
          },
          {
            placeholder: "Company",
            defaultValue: "",
            name: "company",
          },
          {
            placeholder: "City",
            defaultValue: "",
            name: "city",
          }
        ]}
      />
      <CardsWrapper
        attemptGetEntities={attemptGetWorkplaces}
        params={{ workbookId }}
        entities={workplaces}
        getDetails={workplace => [
          {
            value: workplace.get("country"),
            tooltip: "Country",
            icon: "country"
          },
          {
            value: workplace.get("city"),
            tooltip: "City",
            icon: "city"
          },
          {
            value: workplace.get("company"),
            tooltip: "Company",
            icon: "company"
          },
          {
            value: workplace.get("startDate"),
            tooltip: "Start date",
            icon: "date"
          },
          {
            value: workplace.get("endDate"),
            tooltip: "End date",
            icon: "date"
          },
        ]}
        showActions={role === "admin"}
        pagination={pagination}
        isLoading={isLoading}
      >
        {id => (
          <EntityActions
            handleDelete={(event) => {
              event.preventDefault();
              attemptDeleteWorkplace(id)
            }}
            handleEdit={(event) => {
              event.preventDefault();
              history.push(`/workplaces/${workbookId}/${id}/edit`);
            }}
          />
        )}
      </CardsWrapper>
    </React.Fragment>
  );
}

const selectorVariables: ISelectorVariables = {
  workplace: {
    workplaces: true,
    pagination: true,
    isLoading: true
  },
  user: {
    role: true
  }
};

const actions = {
  attemptDeleteWorkplace,
  attemptGetWorkplaces
};

export default connect<IWorkplaceReduxProps, IWorkplaceDispatchProps, any>(mapStateToProps(selectorVariables), mapDispatchToProps(actions))(Workplace)
