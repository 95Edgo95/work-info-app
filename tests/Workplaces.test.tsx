import Workplaces from "../src/containers/Workplaces";
import {BrowserRouter} from "react-router-dom";
import store from "../__mocks__/storeMock";
import {Provider} from "react-redux";
import { mount } from "enzyme";
import * as React from "react";

describe('Workplaces', () => {
  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Workplaces match={{ params: { workbookId: "asdf" } }} />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
