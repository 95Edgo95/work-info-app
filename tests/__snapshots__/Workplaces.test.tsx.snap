// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workplaces should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Workplace)
        match={
          Object {
            "params": Object {
              "workbookId": "asdf",
            },
          }
        }
      >
        <Workplace
          attemptDeleteWorkplace={[Function]}
          attemptGetWorkplaces={[Function]}
          match={
            Object {
              "params": Object {
                "workbookId": "asdf",
              },
            }
          }
          pagination={
            Immutable.Map {
              "count": 0,
              "limit": 0,
              "offset": 0,
            }
          }
          workplaces={Immutable.Map {}}
        >
          <Filter
            additionalParams={
              Object {
                "workbookId": "asdf",
              }
            }
            attemptGetEntities={[Function]}
            createLink="workplaces/asdf"
            inputs={
              Array [
                Object {
                  "defaultValue": "",
                  "name": "country",
                  "placeholder": "Country",
                },
                Object {
                  "defaultValue": "",
                  "name": "company",
                  "placeholder": "Company",
                },
                Object {
                  "defaultValue": "",
                  "name": "city",
                  "placeholder": "City",
                },
              ]
            }
          >
            <div>
              <Input
                key="country"
                name="country"
                onChange={[Function]}
                placeholder="Country"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="country"
                      onChange={[Function]}
                      placeholder="Country"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Input
                key="company"
                name="company"
                onChange={[Function]}
                placeholder="Company"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="company"
                      onChange={[Function]}
                      placeholder="Company"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Input
                key="city"
                name="city"
                onChange={[Function]}
                placeholder="City"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="city"
                      onChange={[Function]}
                      placeholder="City"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Button
                onClick={[Function]}
              >
                <button
                  className=""
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </button>
              </Button>
              <Button
                secondary={true}
              >
                <button
                  className=""
                  type="button"
                >
                  <Link
                    replace={false}
                    to="/workplaces/asdf/create"
                  >
                    <a
                      href="/workplaces/asdf/create"
                      onClick={[Function]}
                    >
                      Add New
                    </a>
                  </Link>
                </button>
              </Button>
            </div>
          </Filter>
          <CardsWrapper
            attemptGetEntities={[Function]}
            entities={Immutable.Map {}}
            getDetails={[Function]}
            pagination={
              Immutable.Map {
                "count": 0,
                "limit": 0,
                "offset": 0,
              }
            }
            params={
              Object {
                "workbookId": "asdf",
              }
            }
          >
            <div>
              <Component />
            </div>
          </CardsWrapper>
        </Workplace>
      </Connect(Workplace)>
    </Router>
  </BrowserRouter>
</Provider>
`;
